src/Api/MediaLibraryApi.php

```diff
@@ -160,14 +160,98 @@ public function mediaPackagePackageNameGet(string $package_name, ?int $limit = 2
    */
   public function mediaFileIdGet(int $id, &$responseCode, array &$responseHeaders)
   {
-    $responseCode = Response::HTTP_I_AM_A_TEAPOT;
+    $json_response_array = [];
+    $media_package_file = $this->entity_manager->getRepository(MediaPackageFile::class)
+      ->findOneBy(['id' => $id])
+        ;
+
+    $responseCode = Response::HTTP_OK; // 200 => OK
+
+    /** @var MediaPackageFile $media_package_file */
+    $mediaFile = [
+      'id' => $media_package_file->getId(),
+      'name' => $media_package_file->getName(),
+      'flavor' => $media_package_file->getFlavor(),
+      'package' => $media_package_file->getCategory()->getPackage()->first()->getName(),
+      'category' => $media_package_file->getCategory()->getName(),
+      'author' => $media_package_file->getAuthor(),
+      'extension' => $media_package_file->getExtension(),
+      'download_url' => $this->url_generator->generate(
+          'download_media',
+          ['id' => $media_package_file->getId()],
+          UrlGenerator::ABSOLUTE_URL),
+    ];
+    $json_response_array[] = new MediaFile($mediaFile);
+
+    $responseData = new MediaFiles(['media_files' => $json_response_array]);
+
+    return $responseData;
   }
 
   /**
    * {@inheritdoc}
    */
   public function mediaFilesGet(?int $limit = 20, ?int $offset = 0, ?string $flavor = null, &$responseCode, array &$responseHeaders)
   {
-    $responseCode = Response::HTTP_I_AM_A_TEAPOT;
+    if (null === $limit)
+    {
+      $limit = 20;
+    }
+    if (null === $offset)
+    {
+      $offset = 0;
+    }
+
+    $total_results = 0;
+
+    $json_response_array = [];
+    $media_package_files = null;
+    if (null == $flavor)
+    {
+      $media_package_files = $this->entity_manager->getRepository(MediaPackageFile::class)
+        ->findAll()
+        ;
+    }
+    else
+    {
+      $media_package_files = $this->entity_manager->getRepository(MediaPackageFile::class)
+        ->findBy(['flavor' => $flavor])
+        ;
+    }
+    if (null !== $media_package_files)
+    {
+      $total_results = count($media_package_files);
```

Comment:
this in combination with all data (1) is very very slow on a large dataset.
Use 2 different queries.

- One to get the data (the query supports and can already use limit + offset ~ no need to reimplement the limit offset logic)
- The second to get the number of total results. Important the query builder should not load all the results only the number (Something like getScalarResults)